packages/circuit-ui/components/SidePanel/SidePanel.tsx

```diff
@@ -0,0 +1,203 @@
+     */
+    onBack?: Callback;
+    /**
+     * Callback function that is called when the side panel is closed.
+     */
+    onClose: Callback;
+    /**
+     * The top offset in 'px' applied to the side panel in desktop mode.
+     */
+    top: string;
+  };
+
+type ContentProps = { top: string };
+
+const contentStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+  padding: 0;
+  padding: env(safe-area-inset-top) env(safe-area-inset-right)
+    env(safe-area-inset-bottom) env(safe-area-inset-left);
+  width: 100%;
+  height: 100%;
+
+  ${theme.mq.mega} {
+    padding-left: 0;
+  }
+`;
+
+const contentTopStyles = ({ top }: ContentProps) =>
+  top !== '0px' &&
+  css`
+    padding-top: 0;
+  `;
+
+const Content = styled.div(contentStyles, contentTopStyles);
+
+const bodyStyles = ({ theme }: StyleProps) => css`
+  width: 100%;
+  max-width: ${BODY_MAX_WIDTH}px;
+  padding: 0 ${theme.spacings.mega};
+
+  ${theme.mq.mega} {
+    padding: 0 ${theme.spacings.giga};
+  }
+`;
+
+const Body = styled.div(bodyStyles);
+
+export const SidePanel = ({
+  backButtonLabel,
+  children,
+  closeButtonLabel,
+  headline,
+  isBottomPanelClosing,
+  isInstantOpen,
+  isMobile,
+  isStacked,
+  onBack,
+  onClose,
+  top,
+  ...props
+}: SidePanelProps): JSX.Element => {
+  if (
+    process.env.UNSAFE_DISABLE_ACCESSIBILITY_ERRORS !== 'true' &&
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test'
+  ) {
+    if (!closeButtonLabel) {
+      throw new Error(
+        'The side panel is missing a `closeButtonLabel` prop. This is an accessibility requirement.',
+      );
+    }
+    if (onBack && !backButtonLabel) {
+      throw new Error(
+        'The side panel is missing a `backButtonLabel` prop. This is an accessibility requirement.',
+      );
+    }
+  }
+
+  const [isHeaderSticky, setHeaderSticky] = useState(false);
+  const headerAriaId = useMemo(() => uniqueId('side-panel-header_'), []);
+
+  useEffect(() => {
+    setHeaderSticky(false);
+  }, [isMobile]);
+
+  const handleScroll: UIEventHandler<HTMLDivElement> = (event) => {
+    setHeaderSticky(event.currentTarget.scrollTop > 0);
+  };
+
+  const defaultProps: Partial<ReactModalProps> = {
+    aria: {
+      labelledby: headerAriaId,
+    },
+    bodyOpenClassName: BODY_OPEN_CLASS_NAME,
+    htmlOpenClassName: HTML_OPEN_CLASS_NAME,
+    onRequestClose: onBack || onClose,
+    portalClassName: PORTAL_CLASS_NAME,
+  };
+
+  const content = (
+    <Content top={top} onScroll={handleScroll}>
+      <Header
+        backButtonLabel={backButtonLabel}
+        closeButtonLabel={closeButtonLabel}
+        headline={headline}
+        id={headerAriaId}
+        onBack={onBack}
+        onClose={onClose}
+        isSticky={isHeaderSticky}
+      />
+      <Body>
+        {isFunction(children) ? children({ onBack, onClose }) : children}
```

Comment:
Neat! I don't think we're doing this for the modal, right? I think we're just accepting a render function like `({ onClose }) => <Children onClose={onClose} />`, and we just don't use the onClose if it's not used: `() => <Children />`

I like this though—maybe we can align the API in other components as well.

Reply:
Actually, I copied this pattern from the Modal component 😄

Reply:
Ah haha 😂 my bad for not remembering this right, then 🙈